tree/ntuple/v7/src/RNTupleMerger.cxx

```diff
@@ -82,26 +82,16 @@ try {
    const int compression =
       mergeInfo->fOptions.Contains("fast") ? kUnknownCompressionSettings : outFile->GetCompressionSettings();
 
-   RNTupleWriteOptions writeOpts;
-   writeOpts.SetUseBufferedWrite(false);
-   if (compression != kUnknownCompressionSettings)
-      writeOpts.SetCompression(compression);
-   auto destination = std::make_unique<RPageSinkFile>(ntupleName, *outFile, writeOpts);
-
-   // If we already have an existing RNTuple, copy over its descriptor to support incremental merging
-   if (outNTuple) {
-      auto source = RPageSourceFile::CreateFromAnchor(*outNTuple);
-      source->Attach();
-      auto desc = source->GetSharedDescriptorGuard();
-      destination->InitFromDescriptor(desc.GetRef());
-   }
-
    // The remaining entries are the input files
    std::vector<std::unique_ptr<RPageSourceFile>> sources;
    std::vector<RPageSource *> sourcePtrs;
 
+   int firstFileCompression = kUnknownCompressionSettings;
    while (const auto &pitr = itr()) {
       TFile *inFile = dynamic_cast<TFile *>(pitr);
+      if (firstFileCompression == kUnknownCompressionSettings)
+         firstFileCompression = inFile->GetCompressionSettings();
```

Comment:
Not sure about this one: since we already diverged the RNTuple compression from the TFile compression (default 505 RNTuple vs default 101 TFile), should we interpret this option rather as: "look at the first _RNTuple_ as a reference for the compression settings".

Reply:
But technically there is no such thing as an "RNTuple compression settings", right? Each column range may in principle have a different compression setting, so which one should we pick?

Reply:
That said, I agree that if I do `-ff` when merging RNTuples, I wouldn't expect the output RNTuple to be compressed with 101 if the input was using the default 505 compression...

Reply:
>  Each column range may in principle have a different compression setting, so which one should we pick?

As a user I would expect each column of the output `RNTuple` to have the same compression setting as they have as in the first input file. (i.e. the output structure/meta-data should be a 'clone' of the structure/meta-data  of the first input `RNTuple`)

Reply:
We might decide that's what we want eventually; I propose that initially we just assume that the first column has the same compression as every other column (since we currently don't expose an API to use different compression).

Reply:
> We might decide that's what we want eventually

Humm ... unless we have a strong reason not to, we ought to be similar to `TTree` which already does the per column copy of the compression setting (implicitly).

Reply:
> What do you mean?

Even though the format supports different compressions per column range, currently from our API you can only set the compression for the entire RNTuple (through `RNTupleWriteOptions`)

Reply:
Does that prevent just the user from customizing the compression settings or does it also prevent the proper implementation inside the merger? [related issue: we are likely to 'forget' to update the Merger when/if we expand the API to allow the per column customization]